go/private/repositories.bzl

```diff
@@ -244,13 +244,13 @@ def go_rules_dependencies(force = False):
     wrapper(
         http_archive,
         name = "org_golang_google_genproto",
-        # main, as of 2023-12-15
+        # main, as of 2024-04-19
         urls = [
-            "https://mirror.bazel.build/github.com/googleapis/go-genproto/archive/995d672761c0c5b9ac6127b488b48825f9a2e5fb.zip",
-            "https://github.com/googleapis/go-genproto/archive/995d672761c0c5b9ac6127b488b48825f9a2e5fb.zip",
+            "https://mirror.bazel.build/github.com/googleapis/go-genproto/archive/8c6c420018be7d99c6336d84554d856523d514bf.zip",
```

Comment:
@linzhp should we be worried about any of these upgrades causing downstream issues?

Reply:
It should be fine as long as `go_rules_dependencies()` is called in WORKSPACE after loading external repos needed by the repo.

However, we should not upgrade any deps in bzlmod unless necessary, because that will force all downstream consumers to upgrade.

With that in mind, @fmeum, do you think we should upgrade WORKSPACE deps as part of every release?

Reply:
On average I would say it's slightly better to keep updating WORKSPACE deps - it makes it less likely that rules_go pulls in an outdated version of something that overrides some other ruleset's deps. But this can still happen when unstable deps contain breaking changes, so it's a mixed bag and we could very well choose to stay out of that.

Reply:
Sounds good. For now let's upgrade them as normal